com.unity.multiplayer.mlapi/Runtime/Core/RPCQueue/QueueHistoryFrame.cs

```diff
@@ -106,28 +107,39 @@ private FrameQueueItem GetCurrentQueueItem()
                 }
             }
 
+            m_CurrentQueueItem.updateStage = m_StreamUpdateStage;
+
             //Get the stream size
             m_CurrentQueueItem.streamSize = queueReader.ReadInt64();
 
-            //Inbound and Outbound message streams are handled differently
-            if (m_QueueFrameType == QueueFrameType.Inbound)
+            //Sanity checking for boundaries
+            if(m_CurrentQueueItem.streamSize < 131072 && m_CurrentQueueItem.streamSize > 0)
             {
-                //Get our offset
-                long Position = queueReader.ReadInt64();
-                //Always make sure we are positioned at the start of the stream before we write
-                m_CurrentQueueItem.itemStream.Position = 0;
+                //Inbound and Outbound message streams are handled differently
+                if (m_QueueFrameType == QueueFrameType.Inbound)
+                {
+                    //Get our offset
+                    long Position = queueReader.ReadInt64();
+                    //Always make sure we are positioned at the start of the stream before we write
+                    m_CurrentQueueItem.itemStream.Position = 0;
 
-                //Write the entire message to the m_CurrentQueueItem stream (1 stream is re-used for all incoming RPCs)
-                m_CurrentQueueItem.streamWriter.ReadAndWrite(queueReader, m_CurrentQueueItem.streamSize);
+                    //Write the entire message to the m_CurrentQueueItem stream (1 stream is re-used for all incoming RPCs)
+                    m_CurrentQueueItem.streamWriter.ReadAndWrite(queueReader, m_CurrentQueueItem.streamSize);
 
-                //Reset the position back to the offset so std rpc API can process the message properly
-                //(i.e. minus the already processed header)
-                m_CurrentQueueItem.itemStream.Position = Position;
+                    //Reset the position back to the offset so std rpc API can process the message properly
+                    //(i.e. minus the already processed header)
+                    m_CurrentQueueItem.itemStream.Position = Position;
+                }
+                else
+                {
+                    //Create a byte array segment for outbound sending
+                    m_CurrentQueueItem.messageData = queueReader.CreateArraySegment((int)m_CurrentQueueItem.streamSize, (int)queueStream.Position);
+                }
             }
             else
             {
-                //Create a byte array segment for outbound sending
-                m_CurrentQueueItem.messageData = queueReader.CreateArraySegment((int)m_CurrentQueueItem.streamSize, (int)queueStream.Position);
+                UnityEngine.Debug.LogWarning("CurrentQueueItem.StreamSize exceeds allowed size (128KB vs " + m_CurrentQueueItem.streamSize.ToString() + ")! Exiting Current RPC Queue Enumeration Loop! ");
```

Comment:
I don't like this warning log for reasons:

- could be just Debug.LogWarning without UnityEngine
  - consider `using UnityEngine;` ?
- there are too many hardcoded strings in there: "CurrentQueueItem.StreamSize", "128KB" etc.
  - consider `nameof`? where 128KB coming from? magic number above?

Reply:
Replaced with the Max stream bounds const, leaving the UnityEngine out of the namespace due to UnityEngine.BitStream conflict.  I could nest the using statements for MLAPI within the namespace to avoid the compilation error =or= I can just reference the UnityEngine namespace directly.